Marlin/src/gcode/calibrate/G33.cpp

```diff
@@ -409,17 +409,21 @@ void GcodeSuite::G33() {
   const bool probe_at_offset = TERN0(HAS_PROBE_XY_OFFSET, parser.boolval('O')),
                   towers_set = !parser.seen_test('T');
 
+  // needs to be moved to all delta config files
+  #define MAX_RADIUS 110
+  
   float max_dcr = dcr = DELTA_PRINTABLE_RADIUS;
   #if HAS_PROBE_XY_OFFSET
-    // For offset probes the calibration radius is set to a safe but non-optimal value
-    dcr -= HYPOT(probe.offset_xy.x, probe.offset_xy.y);
+    // For offset probes the calibration radius is set to a calculated value
     if (probe_at_offset) {
-      // With probe positions both probe and nozzle need to be within the printable area
-      max_dcr = dcr;
+      // with probe positions both probe and nozzle need to be within the printable area
+      max_dcr = dcr -= HYPOT(probe.offset_xy.x, probe.offset_xy.y);
+    }
+    else {
+      // else with nozzle positions only the nozzle needs to be within the printable area
+      // and the probe needs to be inside the physical machine diameter
+      max_dcr = dcr -= _MAX(HYPOT(probe.offset_xy.x, probe.offset_xy.y) - (MAX_RADIUS) + (DELTA_PRINTABLE_RADIUS), 0);
```

Comment:
A probe which is, for example, offset to the left of the nozzle can always reach the left edge, but maybe not the right edge. The `probe.can_reach` method takes this into consideration instead of applying the limit to the entire perimeter as a reduction in radius. Is it important in `G33` for the probing to be symmetrically centered around 0,0 ?

Reply:
1) "Since the probe offsets and the software endstops can both be changed at runtime"

I got that (G33 changes the endstops itself). So having an EEPROM setting for calibration radius is no longer a good idea; That's why I try to calculate it at the start of G33 taking into account the current probe offsets.

2) "The reachable area outside of the bed on a delta might end up being required as a new setting"

I think so as well. Delta calibration is the only occurence were probing on nozzle positions is required, and it's only then that extra margin outside the printable area can be used (and needs to be known).

3) "*_MIN_POS / *_MAX_POS values should have a corresponding delta setting"

*_MIN_POS / *_MAX_POS are quit intensely used by G29 and other stuff and currently are linked to DELTA_PRINTABLE_RADIUS. IMO in order not to disturb the current use of current parameters, it would be better not to link the new parameter to anything since as stated above it's only used for nozzle position probing.

3) "MAX_RADIUS" instead of "BED_DIAMETER"

OK with that. Maybe it would be clearer to users not to refer to this as a radius or diameter but as a a margin like EXTRA_MARGIN_FOR_PROBING, since it only appies to the side the probe is mounted and not to the whole bed.

5) "And, it will be important to give some thought to position_is_reachable and probe.can_reach"

The old testing in G33 was done by testing each point with position_is_reachable; but since it is a circular grid you might as well test dcr against one calculated number.

6) "Is it important in G33 for the probing to be symmetrically centered around 0,0"

For the moment YES. For the match to work (especially with the added probe positions option) it requires symmetry on all 3 axis.